api/v1alpha4/cluster_types.go

```diff
@@ -83,6 +83,7 @@ type Topology struct {
 	RolloutAfter *metav1.Time `json:"rolloutAfter,omitempty"`
 
 	// ControlPlane describes the cluster control plane.
+	// +optional
```

Comment:
@fabriziopandini What should we do here?

Reply:
What if the controlPlane is managed, for example? Then even an empty topology for it doesn't make sense.

Reply:
I don't really have a strong opinion. I just think if the struct is optional (and it is in any case), then it's not a great UX to enforce that users provide an empty struct. When I think about it a bit longer. I think we should maybe also make the field a pointer (same reason, because it's optional to provide it)

Reply:
Looking at the API conventions about optional vs required and comparing them with our structs leads to a bunch of new questions: [optional vs required](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required)
Optional fields:
* They have the +optional comment tag in Go.
* They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) or have a built-in nil value (e.g. maps and slices).
* In most cases, optional fields should also have the omitempty struct tag 

Required fields:
* They do not have an +optional comment tag.
* They do not have an omitempty struct tag.
* They are not a pointer type in the Go definition (e.g. AnotherFlag SomeFlag).

I see a lot of required fields (non-pointer) with omitempty.